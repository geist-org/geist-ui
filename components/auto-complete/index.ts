import AutoComplete from './auto-complete'
import AutoCompleteItem from './auto-complete-item'
import AutoCompleteSearching from './auto-complete-searching'
import AutoCompleteEmpty from './auto-complete-empty'

export type AutoCompleteComponentType = typeof AutoComplete & {
  Item: typeof AutoCompleteItem
  Option: typeof AutoCompleteItem
  Searching: typeof AutoCompleteSearching
  Empty: typeof AutoCompleteEmpty
}
;(AutoComplete as AutoCompleteComponentType).Item = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Option = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Searching = AutoCompleteSearching
;(AutoComplete as AutoCompleteComponentType).Empty = AutoCompleteEmpty

export type {
  AutoCompleteOption,
  AutoCompleteOptions,
  AutoCompleteProps,
  AutoCompleteTypes,
} from './auto-complete'
export default AutoComplete as AutoCompleteComponentType
