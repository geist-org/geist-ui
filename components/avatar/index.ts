import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
