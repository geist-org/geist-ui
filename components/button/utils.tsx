import React, { ReactNode } from 'react'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'
import useClasses from '../use-classes'

export const getButtonChildrenWithIcon = (
  auto: boolean,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode = auto
    ? `calc(var(--geist-ui-button-height) / 2 + var(--geist-ui-button-icon-padding) * .5)`
    : 0
  const classes = useClasses('text', isRight ? 'right' : 'left')

  if (!hasIcon) return <div className="text">{children}</div>
  if (React.Children.count(children) === 0) {
    return (
      <ButtonIcon isRight={isRight} isSingle>
        {hasIcon}
      </ButtonIcon>
    )
  }
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={classes}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = <T extends React.PropsWithChildren<ButtonProps>>(
  props: T,
  config: ButtonGroupConfig,
): T => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    type: config.type || props.type,
    disabled: config.disabled || props.disabled,
  }
}
