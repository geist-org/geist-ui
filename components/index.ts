// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
