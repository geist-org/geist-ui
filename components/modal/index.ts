import Modal from './modal'
import ModalTitle from './modal-title'
import ModalSubtitle from './modal-subtitle'
import ModalContent from './modal-content'
import ModalAction from './modal-action'

export type ModalComponentType = typeof Modal & {
  Title: typeof ModalTitle
  Subtitle: typeof ModalSubtitle
  Content: typeof ModalContent
  Action: typeof ModalAction
}
;(Modal as ModalComponentType).Title = ModalTitle
;(Modal as ModalComponentType).Subtitle = ModalSubtitle
;(Modal as ModalComponentType).Content = ModalContent
;(Modal as ModalComponentType).Action = ModalAction

export type { ModalProps } from './modal'
export type { ModalTitleProps } from './modal-title'
export type { ModalSubtitleProps } from './modal-subtitle'
export type { ModalActionProps } from './modal-action'
export type { ModalContentProps } from './modal-content'
export default Modal as ModalComponentType
