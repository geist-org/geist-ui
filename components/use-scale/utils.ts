import {
  GetAllScalePropsFunction,
  GetScalePropsFunction,
  ScaleProps,
  ScalePropKeys,
} from './scale-context'

export const generateGetScaleProps = <P>(
  props: P & ScaleProps,
): GetScalePropsFunction => {
  const getScaleProps: GetScalePropsFunction = keyOrKeys => {
    if (!Array.isArray(keyOrKeys)) return props[keyOrKeys as keyof ScaleProps]
    let value = undefined
    for (const key of keyOrKeys) {
      const currentValue = props[key]
      if (typeof currentValue !== 'undefined') {
        value = currentValue
      }
    }
    return value
  }
  return getScaleProps
}

export const generateGetAllScaleProps = <P>(
  props: P & ScaleProps,
): GetAllScalePropsFunction => {
  const getAllScaleProps: GetAllScalePropsFunction = () => {
    let scaleProps: ScaleProps = {}
    for (const key of ScalePropKeys) {
      const value = props[key as keyof ScaleProps]
      if (typeof value !== 'undefined') {
        scaleProps[key as keyof ScaleProps] = value as any
      }
    }
    return scaleProps
  }
  return getAllScaleProps
}
