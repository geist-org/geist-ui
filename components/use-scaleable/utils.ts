import { ScaleableProps } from './scaleable-context'

export type ScaleablePropsAndInvalid = keyof ScaleableProps | 'size'

export const ScaleablePropKeys: Array<ScaleablePropsAndInvalid> = [
  'paddingLeft',
  'pl',
  'paddingRight',
  'pr',
  'paddingTop',
  'pt',
  'paddingBottom',
  'pb',
  'marginTop',
  'mt',
  'marginRight',
  'mr',
  'marginBottom',
  'mb',
  'marginLeft',
  'ml',
  'px',
  'py',
  'mx',
  'my',
  'width',
  'height',
  'font',
  'unit',
  'scale',
  'size',
]

export const filterScaleableProps = <T extends Record<any, any>>(props: T) => {
  const keys = Object.keys(props).filter(key => key !== '')
  const nextProps: any = {}
  for (const key of keys) {
    if (!(ScaleablePropKeys as string[]).includes(key)) {
      nextProps[key] = props[key]
    }
  }
  return nextProps as Omit<T, ScaleablePropsAndInvalid>
}
