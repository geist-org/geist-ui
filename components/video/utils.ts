export const formatTime = (seconds: number) => {
  const hh = Math.floor(seconds / 3600)
  const mm = Math.floor(seconds / 60) % 60
  const ss = Math.floor(seconds) % 60
  const pad = (num: number) => (num < 10 ? '0' + num : num)
  return `${hh ? pad(hh) + ':' : ''}${pad(mm)}:${pad(ss)}`
}

export const isFullscreenEnabled = () => {
  const elem = document as any
  return (
    elem.fullscreenEnabled ||
    elem.webkitFullscreenEnabled ||
    elem.mozFullScreenEnabled ||
    elem.msFullscreenEnabled
  )
}

export const openFullscreen = (element: HTMLElement) => {
  const elem = element as any
  if (elem.requestFullscreen) {
    elem.requestFullscreen()
  } else if (elem.mozRequestFullScreen) {
    elem.mozRequestFullScreen()
  } else if (elem.webkitRequestFullscreen) {
    elem.webkitRequestFullscreen()
  } else if (elem.msRequestFullscreen) {
    elem.msRequestFullscreen()
  }
}

export const closeFullscreen = () => {
  const elem = document as any
  if (elem.exitFullscreen) {
    elem.exitFullscreen()
  } else if (elem.mozCancelFullScreen) {
    elem.mozCancelFullScreen()
  } else if (elem.webkitExitFullscreen) {
    elem.webkitExitFullscreen()
  } else if (elem.msExitFullscreen) {
    elem.msExitFullscreen()
  }
}

export const onFullscreenChange = (func: (state: boolean) => void) => {
  const doc = document as any
  doc.addEventListener('fullscreenchange', () => func(!!doc.fullscreen), true)
  doc.addEventListener('mozfullscreenchange', () => func(!!doc.mozFullScreen), true)
  doc.addEventListener('webkitfullscreenchange', () => func(!!doc.webkitIsFullScreen), true)
  doc.addEventListener('msfullscreenchange', () => func(!!doc.msFullscreenElement), true)
}

export const isSafari = () =>
  navigator.userAgent.includes('Safari') && !navigator.userAgent.includes('Chrome')
