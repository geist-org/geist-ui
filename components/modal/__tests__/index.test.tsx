import React from 'react'
import { mount } from 'enzyme'
import { Modal } from 'components'
import { nativeEvent, updateWrapper } from 'tests/utils'
import {
  expectModalIsClosed,
  expectModalIsOpened,
} from '../../use-modal/__tests__/use-modal.test'
import userEvent from '@testing-library/user-event'

describe('Modal', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Modal visible={true}>
        <Modal.Title>Modal</Modal.Title>
        <Modal.Subtitle>This is a modal</Modal.Subtitle>
        <Modal.Content>
          <p>Some content contained within the modal.</p>
        </Modal.Content>
        <Modal.Action passive>Cancel</Modal.Action>
        <Modal.Action>Submit</Modal.Action>
      </Modal>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should trigger event when modal changed', async () => {
    const closeHandler = jest.fn()
    const wrapper = mount(
      <Modal onClose={closeHandler}>
        <Modal.Title>Modal</Modal.Title>
      </Modal>,
    )
    expectModalIsClosed(wrapper)

    wrapper.setProps({ visible: true })
    await updateWrapper(wrapper, 350)
    expectModalIsOpened(wrapper)

    wrapper.find('.backdrop').simulate('click', nativeEvent)
    await updateWrapper(wrapper, 500)
    expectModalIsClosed(wrapper)
    expect(closeHandler).toHaveBeenCalled()
  })

  it('should disable backdrop event', async () => {
    const closeHandler = jest.fn()
    const wrapper = mount(
      <Modal visible={true} disableBackdropClick onClose={closeHandler}>
        <Modal.Title>Modal</Modal.Title>
        <Modal.Action>Submit</Modal.Action>
      </Modal>,
    )
    wrapper.find('.backdrop').simulate('click', nativeEvent)
    await updateWrapper(wrapper, 500)
    expectModalIsOpened(wrapper)
    expect(closeHandler).not.toHaveBeenCalled()
  })

  it('should disable backdrop even if actions missing', async () => {
    const closeHandler = jest.fn()
    const wrapper = mount(
      <Modal visible={true} disableBackdropClick onClose={closeHandler}>
        <Modal.Title>Modal</Modal.Title>
      </Modal>,
    )
    wrapper.find('.backdrop').simulate('click', nativeEvent)
    await updateWrapper(wrapper, 500)
    expectModalIsOpened(wrapper)
    expect(closeHandler).not.toHaveBeenCalled()
  })

  it('should ignore event when action disabled', () => {
    const actions1 = jest.fn()
    const actions2 = jest.fn()
    const wrapper = mount(
      <Modal visible={true}>
        <Modal.Title>Modal</Modal.Title>
        <Modal.Action passive onClick={actions1}>
          Submit
        </Modal.Action>
        <Modal.Action disabled onClick={actions2}>
          Submit
        </Modal.Action>
      </Modal>,
    )
    wrapper.find('button').at(0).simulate('click', nativeEvent)
    wrapper.find('button').at(1).simulate('click', nativeEvent)

    expect(actions1).toHaveBeenCalled()
    expect(actions2).not.toHaveBeenCalled()
  })

  it('should be close modal through action event', async () => {
    const closeHandler = jest.fn()
    const wrapper = mount(
      <Modal visible={true} onClose={closeHandler}>
        <Modal.Title>Modal</Modal.Title>
        <Modal.Action passive onClick={e => e.close()}>
          Close
        </Modal.Action>
      </Modal>,
    )
    wrapper.find('button').at(0).simulate('click', nativeEvent)
    await updateWrapper(wrapper, 500)
    expectModalIsClosed(wrapper)
    expect(closeHandler).toHaveBeenCalled()
  })

  it('customization should be supported', () => {
    const wrapper = mount(
      <Modal visible={true} width="100px" wrapClassName="test-class">
        <Modal.Title>Modal</Modal.Title>
      </Modal>,
    )
    const html = wrapper.find('.wrapper').html()
    expect(html).toMatchSnapshot()
    expect(wrapper.find('.wrapper').at(0).getDOMNode()).toHaveClass('test-class')
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('focus should only be switched within modal', async () => {
    const wrapper = mount(
      <Modal visible={true} width="100px" wrapClassName="test-class">
        <button id="button" />
      </Modal>,
    )
    const tabStart = wrapper.find('.hide-tab').at(0).getDOMNode()
    const tabEnd = wrapper.find('.hide-tab').at(1).getDOMNode()
    const button = wrapper.find('#button').at(0).getDOMNode()
    const focusTrap = wrapper.find('.wrapper').at(0).getDOMNode()

    expect(tabStart).toHaveFocus()
    userEvent.tab({ focusTrap })
    expect(button).toHaveFocus()
    userEvent.tab()
    expect(tabEnd).toHaveFocus()
    userEvent.tab()
    expect(tabStart).toHaveFocus()

    userEvent.tab({ shift: true, focusTrap })
    expect(tabEnd).toHaveFocus()
    userEvent.tab({ shift: true, focusTrap })
    expect(button).toHaveFocus()
    userEvent.tab({ shift: true, focusTrap })
    expect(tabStart).toHaveFocus()
  })

  it('should close modal when keyboard event is triggered', async () => {
    const wrapper = mount(
      <Modal visible={true}>
        <Modal.Title>Modal</Modal.Title>
      </Modal>,
    )
    expectModalIsOpened(wrapper)
    userEvent.keyboard('{esc}')
    await updateWrapper(wrapper, 500)
    expectModalIsClosed(wrapper)
  })

  it('should prevent close modal when keyboard is false', async () => {
    const wrapper = mount(
      <Modal visible={true} keyboard={false}>
        <Modal.Title>Modal</Modal.Title>
      </Modal>,
    )
    expectModalIsOpened(wrapper)
    userEvent.keyboard('{esc}')
    await updateWrapper(wrapper, 500)
    expectModalIsOpened(wrapper)
  })
})
