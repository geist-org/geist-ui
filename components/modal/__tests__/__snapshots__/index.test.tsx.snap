// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal customization should be supported 1`] = `
"<div class=\\"wrapper test-class  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab\\" aria-hidden=\\"true\\"></div><h2 class=\\"\\">Modal</h2><style>
        h2 {
          line-height: 1.6;
          font-weight: normal;
          text-align: center;
          display: inline-flex;
          flex-shrink: 0;
          justify-content: center;
          align-items: center;
          word-break: break-word;
          text-transform: capitalize;
          font-size: calc(1.5 * 16px);
          color: #000;
          width: auto;
          height: auto;
          padding: 0 0 0 0;
          margin: 0 0 0 0;
        }
      </style><div tabindex=\\"0\\" class=\\"hide-tab\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            max-width: 100%;
            vertical-align: middle;
            overflow: hidden;
            display: flex;
            flex-direction: column;
            position: relative;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: 6px;
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(0px, -30px, 0px);
            transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1) 0s,
              transform 0.35s cubic-bezier(0.4, 0, 0.2, 1) 0s;
            width: 100%;
            font-size: calc(1 * 16px);
            height: auto;
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }

          .wrapper-enter {
            opacity: 0;
            transform: translate3d(0px, -30px, 0px);
          }

          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0px, 0px, 0px);
          }

          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0px, 0px, 0px);
          }

          .wrapper-leave-active {
            opacity: 0;
            transform: translate3d(0px, -30px, 0px);
          }

          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div>"
`;

exports[`Modal should render correctly 1`] = `
"<div class=\\"backdrop  backdrop-wrapper-enter\\"><div class=\\"layer\\"></div><div class=\\"position\\"><div class=\\"wrapper  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab\\" aria-hidden=\\"true\\"></div><h2 class=\\"\\">Modal</h2><style>
        h2 {
          line-height: 1.6;
          font-weight: normal;
          text-align: center;
          display: inline-flex;
          flex-shrink: 0;
          justify-content: center;
          align-items: center;
          word-break: break-word;
          text-transform: capitalize;
          font-size: calc(1.5 * 16px);
          color: #000;
          width: auto;
          height: auto;
          padding: 0 0 0 0;
          margin: 0 0 0 0;
        }
      </style><p class=\\"\\">This is a modal</p><style>
        p {
          font-weight: normal;
          display: inline-block;
          text-align: center;
          word-break: break-word;
          text-transform: uppercase;
          color: #666;
          font-size: calc(0.875 * 16px);
          line-height: 1.5em;
          width: auto;
          height: 1.5em;
          padding: 0 0 0 0;
          margin: 0 0 0 0;
        }
      </style><div class=\\"content\\"><p>Some content contained within the modal.</p></div><style>
        .content {
          position: relative;
          text-align: left;
          font-size: calc(1 * 16px);
          width: auto;
          height: auto;
          padding: calc(1.3125 * 16px) calc(1.3125 * 16px) calc(0.6625 * 16px)
            calc(1.3125 * 16px);
          margin: 0
            calc(var(--modal-wrapper-padding-right) * -1)
            0
            calc(var(--modal-wrapper-padding-left) * -1);
        }

        .content > :global(*:first-child) {
          margin-top: 0;
        }

        .content > :global(*:last-child) {
          margin-bottom: 0;
        }
      </style><div></div><footer><button type=\\"button\\" class=\\"btn mock\\"><div class=\\"text\\">Cancel</div><style>
          .btn {
            box-sizing: border-box;
            display: inline-block;
            line-height: calc(2.5 * 16px);
            border-radius: 6px;
            font-weight: 400;
            font-size: calc(0.875 * 16px);
            user-select: none;
            outline: none;
            text-transform: capitalize;
            justify-content: center;
            text-align: center;
            white-space: nowrap;
            transition: background-color 200ms ease 0ms, box-shadow 200ms ease 0ms,
              border 200ms ease 0ms, color 200ms ease 0ms;
            position: relative;
            overflow: hidden;
            color: #666;
            background-color: #fff;
            border: 1px solid #eaeaea;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            --geist-ui-button-icon-padding: calc(0.727 * 16px);
            --geist-ui-button-height: calc(2.5 * 16px);
            --geist-ui-button-color: #666;
            --geist-ui-button-bg: #fff;
            min-width: calc(10.5 * 16px);
            width: initial;
            height: calc(2.5 * 16px);
            padding: 0 calc(1.375 * 16px) 0 calc(1.375 * 16px);
            margin: 0 0 0 0;
          }

          .btn:hover,
          .btn:focus {
            color: #000;
            --geist-ui-button-color: #000;
            background-color: #fff;
            border-color: #000;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            transform: translate3d(0px, 0px, 0px);
          }

          .btn :global(.text) {
            position: relative;
            z-index: 1;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            text-align: center;
            line-height: inherit;
            top: -1px;
          }

          .btn :global(.text p),
          .btn :global(.text pre),
          .btn :global(.text div) {
            margin: 0;
          }
        </style></button><button type=\\"button\\" class=\\"btn mock\\"><div class=\\"text\\">Submit</div><style>
          .btn {
            box-sizing: border-box;
            display: inline-block;
            line-height: calc(2.5 * 16px);
            border-radius: 6px;
            font-weight: 400;
            font-size: calc(0.875 * 16px);
            user-select: none;
            outline: none;
            text-transform: capitalize;
            justify-content: center;
            text-align: center;
            white-space: nowrap;
            transition: background-color 200ms ease 0ms, box-shadow 200ms ease 0ms,
              border 200ms ease 0ms, color 200ms ease 0ms;
            position: relative;
            overflow: hidden;
            color: #666;
            background-color: #fff;
            border: 1px solid #eaeaea;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            --geist-ui-button-icon-padding: calc(0.727 * 16px);
            --geist-ui-button-height: calc(2.5 * 16px);
            --geist-ui-button-color: #666;
            --geist-ui-button-bg: #fff;
            min-width: calc(10.5 * 16px);
            width: initial;
            height: calc(2.5 * 16px);
            padding: 0 calc(1.375 * 16px) 0 calc(1.375 * 16px);
            margin: 0 0 0 0;
          }

          .btn:hover,
          .btn:focus {
            color: #000;
            --geist-ui-button-color: #000;
            background-color: #fff;
            border-color: #000;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            transform: translate3d(0px, 0px, 0px);
          }

          .btn :global(.text) {
            position: relative;
            z-index: 1;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            text-align: center;
            line-height: inherit;
            top: -1px;
          }

          .btn :global(.text p),
          .btn :global(.text pre),
          .btn :global(.text div) {
            margin: 0;
          }
        </style></button></footer><style>
        footer {
          display: flex;
          overflow: hidden;
          width: 100%;
          height: auto;
          position: absolute;
          bottom: 0;
          left: 0;
          right: 0;
          border-top: 1px solid #eaeaea;
          border-bottom-left-radius: 6px;
          border-bottom-right-radius: 6px;
        }

        footer > :global(button.btn + button.btn) {
          border-left: 1px solid #eaeaea;
        }

        div {
          height: 0px;
          flex-shrink: 0;
        }
      </style><div tabindex=\\"0\\" class=\\"hide-tab\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            max-width: 100%;
            vertical-align: middle;
            overflow: hidden;
            display: flex;
            flex-direction: column;
            position: relative;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: 6px;
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(0px, -30px, 0px);
            transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1) 0s,
              transform 0.35s cubic-bezier(0.4, 0, 0.2, 1) 0s;
            width: 100%;
            font-size: calc(1 * 16px);
            height: auto;
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }

          .wrapper-enter {
            opacity: 0;
            transform: translate3d(0px, -30px, 0px);
          }

          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0px, 0px, 0px);
          }

          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0px, 0px, 0px);
          }

          .wrapper-leave-active {
            opacity: 0;
            transform: translate3d(0px, -30px, 0px);
          }

          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div></div><style>
            .backdrop {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              overflow: auto;
              z-index: 1000;
              -webkit-overflow-scrolling: touch;
              box-sizing: border-box;
              text-align: center;
            }
            .position {
              position: relative;
              z-index: 1001;
              outline: none;
              max-width: 90%;
              width: calc(26 * 16px);
              margin: 20px auto;
              vertical-align: middle;
              display: inline-block;
            }
            .backdrop:before {
              display: inline-block;
              width: 0;
              height: 100%;
              vertical-align: middle;
              content: '';
            }
            .layer {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              width: 100%;
              height: 100%;
              opacity: 0.25;
              background-color: black;
              transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1);
              pointer-events: none;
              z-index: 1000;
            }
            .backdrop-wrapper-enter .layer {
              opacity: 0;
            }
            .backdrop-wrapper-enter-active .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave-active .layer {
              opacity: 0;
            }
          </style></div>"
`;
